### ISSUE TITLE
When querying pyarrow dataset, all columns are retrieved when no columns are needed

### ISSUE THREAD

**Issue by westonpace** [duckdb/duckdb#14718]
### What happens?

If I issue the query `SELECT COUNT(1) FROM my_dataset` then all columns will be loaded from disk.

This is because duckdb does not set the `columns` property when calling `Dataset.scanner` which pyarrow interprets as "load all columns".  It would be more efficient to pass in `columns=[]`.

### To Reproduce

```
import pyarrow as pa
import pyarrow.dataset as ds

import duckdb

# Flimsy dataset wrapper to tell us how DuckDb is using the pyarrow dataset                                                                                                                                        
class DuckDataset(ds.Dataset):
    def __init__(self, ds):
	self._ds = ds

    @property
    def schema(self):
	return self._ds.schema

    def scanner(self, *args, **kwargs):
        if "columns" in kwargs:
            print(f"columns={kwargs['columns']}")
        else:
            print("columns not set")
	return self._ds.scanner(*args, **kwargs)

tab = pa.table({"x": [1, 2, 3, 4, 5], "y": [1, 2, 3, 4, 5]})
ds = DuckDataset(ds.dataset(tab))

conn = duckdb.connect()
# Fetches all columns (bad)                                                                                                                                                                                        
print(conn.sql("SELECT COUNT(1) FROM ds").fetchall())
# Only fetches column x (good)                                                                                                                                                                                     
print(conn.sql("SELECT COUNT(x) FROM ds").fetchall())

# A more efficient way to calculate the number of rows                                                                                                                                                             
num_rows = 0
for batch in ds.scanner(columns=[]).to_batches():
    num_rows += batch.num_rows
print(f"Total num rows: {num_rows}")
```

### OS:

Ubuntu

### DuckDB Version:

1.1.3

### DuckDB Client:

Python

### Hardware:

N/A

### Full Name:

Weston Pace

### Affiliation:

LanceDB

### What is the latest build you tested with? If possible, we recommend testing with the latest nightly build.

I have tested with a stable release

### Did you include all relevant data sets for reproducing the issue?

Not applicable - the reproduction does not require a data set

### Did you include all code required to reproduce the issue?

- [X] Yes, I have

### Did you include all relevant configuration (e.g., CPU architecture, Python version, Linux distribution) to reproduce the issue?

- [X] Yes, I have

**Comment by Tishj** [comment]
Thanks, can you provide sources to reference for "`columns=[]` A more efficient way to calculate the number of rows" ?
That sounds like it doesn't fetch any columns

**Comment by westonpace** [comment]
> That sounds like it doesn't fetch any columns

This is correct.  The query I am hoping to optimize is `SELECT COUNT(1) FROM ds`.  This query should not ideally not have to fetch any columns.  In some cases (parquet / orc) pyarrow can do this efficiently by only looking at the metadata.  In other cases (csv) then some I/O is still required to count the number of rows.

There are several ways the optimal path can be achieved with a pyarrow dataset:

* One option would be to call `count_rows` on the dataset (https://arrow.apache.org/docs/python/generated/pyarrow.dataset.Dataset.html#pyarrow.dataset.Dataset.count_rows)
* Another option is to call `count_rows` on the returned scanner (https://arrow.apache.org/docs/python/generated/pyarrow.dataset.Scanner.html#pyarrow.dataset.Scanner.count_rows).  This is what `Dataset.count_rows` is doing (https://github.com/apache/arrow/blob/main/python/pyarrow/_dataset.pyx#L776)
* Finally, another option is to pass in `columns=[]`.  I don't have a great reference for this, and it has been some time since I actively worked on the scanner, and so things may have changed, but I would be surprised if this triggered a read of all the data and I can confirm it returns batches with 0 columns which suggests it is not actually reading data.

So I suppose the first two options would be safer, but my guess is they would require more logic in the integration.

**Comment by szarnyasg** [comment]
@westonpace thanks for opening this issue. I get the following output when running your script:

```console
columns not set
[(5,)]
columns=['x']
Traceback (most recent call last):
  File "/Users/szarnyasg/tmp/my.py", line 29, in <module>
    print(conn.sql("SELECT COUNT(x) FROM ds").fetchall())
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
duckdb.duckdb.Error: AttributeError: 'NoneType' object has no attribute 'to_reader'
```

**Comment by westonpace** [comment]
@szarnyasg ah, had in error in transcription to github.  Script has been updated.
